### PR TITLE
Find route around barriers

### PR DESCRIPTION
**Pull request by prupani-7**
## Description
PR to add a new Kotlin sample "Find route around barriers" in Routes and Logistics category. This sample finds a route that reaches all stops without crossing any barriers.

## Links and Data
Sample Epic: runtime/kotlin/issues/1695 

## What To Review

-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

### CODE REVIEW

**Comment by shubham7109** [comment]
Hopefully, you don't mind but I've updated the dialog to use view binding instead of the `findViewById<>`. And updated the switches to persist the state of the toggle to match the iOS sample.

**Review by shubham7109 — CHANGES_REQUESTED** [review]
Really impressive work with this sample @prupani-7. This sample is a great app to showcase the capabilities of the Maps SDK when many aspects of the API are interacting together in relation to the Android views and the MapView. I just have a couple of comments, but really well-done sample here.

**Comment by prupani-7** [comment]
> Hopefully, you don't mind but I've updated the dialog to use view binding instead of the `findViewById<>`. And updated the switches to persist the state of the toggle to match the iOS sample.

Thanks, yes it is better to use view binding and also have all switches initialized at the start. 
I refactored the switches code, to remove redundancy while checking `findBestSequenceSwitch.isChecked`. It was done twice. Also set the `isChecked = false` on `firstStopSwitch` and `lastStopSwitch` when `findBestSequenceSwitch` is unchecked.

File: find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt
Comment (prupani-7) [reply find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt]: renamed to routeResults

**Review by shubham7109 — APPROVED** [review]
Changes look good 👍

File: find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt
Comment (prupani-7) [reply find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt:325]: Sounds good.

File: find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt
Comment (fpearson1111) [inline find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt:408]: I think it's better to just use one `let` here because the result of the first `let` will never be null, only if the geometry is null.
```suggestion
                        geometry?.let { mapView.setViewpoint(Viewpoint(it.extent, 20.0)) }
```

File: find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt
Comment (fpearson1111) [inline find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt:335]: Is this check still needed?

**Review by fpearson1111 — COMMENTED** [review]
Sorry for the long delay in completing the first round of this review. I've gone through the full code now and it looks good except just a few comments.

Although I also tried the sample and for me the directions list isn't displaying anything. Can you try it out again on your device?

**Comment by prupani-7** [comment]
Thanks @fpearson1111 no worries!
yes I am able to see the directions list on my device
image
I tap on the map twice, that creates 2 points and displays a route between them and the directions too

**Review by fpearson1111 — APPROVED** [review]
I think the reason I wasn't seeing any directions is that my phone is in dark mode. When I switch it to light mode the instructions appear. I attached some screenshots:



I think this sample should be updated so that the instructions still display in dark mode but I will approve anyway because if the fix is easy then I am happy for it to be merged without re-approval. Nice work :)

**Comment by prupani-7** [comment]
I think that is something, we will need to update on the samples, which display directions list. There are 2 more samples, which will need an update I think, which displays the directions text. I will create a follow up issue, because I am not sure, how much the scope of work will be at this moment. 

The three samples which will need an update will be 
`find route around barriers`
`navigate route`
`find-route`

**Comment by prupani-7** [comment]
Thanks for the feedback @fpearson1111 always helpful!

File: find-route-around-barriers/src/main/res/values/styles.xml
Comment (shubham7109) [inline find-route-around-barriers/src/main/res/values/styles.xml:4]:
```suggestion
    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar" >
```
For now, samples should be set to be in light theme, we can take on supporting dark theme in a future issue.

Comment (shubham7109) [inline find-route-around-barriers/src/main/res/values/styles.xml:4]:
```suggestion
    <style name="AppTheme" parent="Theme.MaterialComponents.Light.DarkActionBar" >
```
For now, samples should be set to be in light theme, we can take on supporting dark theme in a future issue.

Comment (prupani-7) [reply find-route-around-barriers/src/main/res/values/styles.xml:4]: Thanks @shubham7109 makes sense!

**Pull request merged**
